src/main/java/com/spring/rest/model/mappers/CarCharacteristicsMapper.java

```diff
@@ -0,0 +1,71 @@
+package com.spring.rest.model.mappers;
+
+import com.spring.rest.model.CarCharacteristics;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Component;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Used to create CarCharacteristics from data obtained from database.
+ */
+@Component
+public class CarCharacteristicsMapper implements RowMapper<CarCharacteristics> {
+    /**
+     * The constant CAR_ID.
+     */
+    public static final String CAR_ID = "car_id";
+    /**
+     * The constant YEAR.
+     */
+    public static final String YEAR = "year";
+    /**
+     * The constant MILEAGE.
+     */
+    public static final String MILEAGE = "mileage";
+    /**
+     * mapper to get FuelType object.
+     */
+    private FuelTypeMapper fuelTypeMapper;
+    /**
+     * mapper to get Transmission object.
+     */
+    private TransmissionMapper transmissionMapper;
+    /**
+     * mapper to get CarModel object.
+     */
+    private CarModelMapper carModelMapper;
+    /**
+     * Logger.
+     */
+    private static final Logger LOGGER = LogManager.getLogger(CarCharacteristicsMapper.class);
+
+    /**
+     * Instantiates a new CarCharacteristics mapper.
+     *
+     * @param fuelTypeMapper     the fuel type mapper
+     * @param transmissionMapper the transmission mapper
+     * @param carModelMapper     the car model mapper
+     */
+    @Autowired
+    public CarCharacteristicsMapper(final FuelTypeMapper fuelTypeMapper, final TransmissionMapper transmissionMapper,
+                                    final CarModelMapper carModelMapper) {
+        this.fuelTypeMapper = fuelTypeMapper;
+        this.transmissionMapper = transmissionMapper;
+        this.carModelMapper = carModelMapper;
+    }
+
+
+    @Override
+    public CarCharacteristics mapRow(final ResultSet resultSet, final int i) throws SQLException {
```

Comment:
What is `i`?

Reply:
It is a method of the RowMapper interface.
I've just override it.  
the parameter i is the number of the current row

Reply:
Why didn't you use it?

Reply:
It is not necessary for me, because I set a specific value to a specific field
parameter 'i' just shows the current row's number